flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointIDCounterTestBase.java

```diff
@@ -116,23 +114,21 @@ public void testConcurrentGetAndIncrement() throws Exception {
             // Get the counts
             for (Future<List<Long>> result : resultFutures) {
                 List<Long> counts = result.get();
+                assertStrictlyMonotonous(counts);
                 all.addAll(counts);
             }
 
             // Verify
             Collections.sort(all);
 
-            assertEquals(expectedTotal, all.size());
+            assertThat(all.size()).isEqualTo(expectedTotal);
 
-            long current = 0;
-            for (long val : all) {
-                // Incrementing counts
-                assertEquals(++current, val);
-            }
+            assertStrictlyMonotonous(all);
 
             // The final count
-            assertEquals(expectedTotal + 1, counter.get());
-            assertEquals(expectedTotal + 1, counter.getAndIncrement());
+            final long lastCheckpointId = all.get(all.size() - 1);
+            assertThat(lastCheckpointId).isLessThan(counter.get());
+            assertThat(lastCheckpointId).isLessThan(counter.getAndIncrement());
```

Comment:
```suggestion
            assertThat(all).last(LONG).isLessThan(counter.get());
            assertThat(all).last(LONG).isLessThan(counter.getAndIncrement());
```
just playing around with the assertj API a bit here. Doesn't need to be applied. 8)